### PR TITLE
Updates needed for substrate #4820 (composite accounts)

### PR DESCRIPTION
**Pull request by gavofyork**

### CODE REVIEW

**Comment by jacogr** [comment]
Tested this against an API with the `type Address = AccountId` and transfers as well as remarks works as expected. Checked the transactions -

- "from" address passed as an AccountId (32-bytes )works as expected (in unchecked)
- using AccountId as lookup source (e.g. transfer "to") works as expected
- data shows up under system.account (balances)

So functionally it looks correct from my early testing. Will run through more.

File: runtime/common/src/claims.rs [hunk 1]
```diff
@@ -161,31 +161,23 @@ decl_module! {
 			let balance_due = <Claims<T>>::get(&signer)
 				.ok_or(Error::<T>::SignerHasNoClaim)?;
 
-			let maybe_vested = <Vesting<T>>::get(&signer);
-
-			// If this fails, destination account already has a vesting schedule
-			// applied to it, and this claim should not be processed.
-			ensure!(
```
Comment (shawntabrizi) [inline runtime/common/src/claims.rs]: When you removed this ensure check, I also believe you now made it possible for someone with a vesting schedule to drop the pot to zero by calling this function over and over.

Comment (shawntabrizi) [reply runtime/common/src/claims.rs]: I have updated the logic and added a test which would have failed if this logic didn't exist.

https://github.com/paritytech/polkadot/pull/834/commits/15fe7cc4008582cd630fa564204115210c5de258

File: network/src/legacy/gossip/mod.rs [hunk 2]
```diff
@@ -283,11 +283,19 @@ pub fn register_validator<C: ChainContext + 'static, S: NetworkSpecialization<Bl
...
 
+	// Ideally this would not be spawned as an orphaned task, but polled by
+	// `RegisteredMessageValidator` which in turn would be polled by a `ValidationNetwork`.
+	let spawn_res = executor.spawn_obj(futures::task::FutureObj::from(Box::new(gossip_engine.clone())));
+
+	// Note: we consider the chances of an error to spawn a background task almost null.
+	if spawn_res.is_err() {
+		log::error!(target: "polkadot-gossip", "Failed to spawn background task");
+	}
+
```
Comment (shawntabrizi) [inline network/src/legacy/gossip/mod.rs]: Can someone else who knows that this does give it a thumbs up?

**Comment by NikVolf** [comment]
I've included changes in #848 853

**Comment by andresilva** [comment]
Superseded by #853.

**Pull request closed without merging**